src/httpserver.cpp

```diff
@@ -459,14 +475,22 @@ void StopHTTPServer()
         evhttp_del_accept_socket(eventHTTP, socket);
     }
     boundSockets.clear();
+    {
+        WAIT_LOCK(g_requests_mutex, lock);
+        while (!g_requests.empty()) {
+            g_requests_cv.wait(lock);
+        }
+    }
+    if (eventHTTP) {
+        event_base_once(eventBase, -1, EV_TIMEOUT, [](evutil_socket_t, short, void*) {
```

Comment:
ea724b9 curious why you added this line

The other place in this codebase that calls `event_base_once()` to schedule a one-time event is torcontrol.cpp in `InterruptTorControl`.

Reply:
This is a way to call `evhttp_free` in the event base thread, meaning it's not called concurrently with requests.